cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -170,16 +170,18 @@ public void apply() throws Exception {
     }
   }
 
-  private void executePersist(final TriggerKey triggerKey, final Trigger.TriggerState state) {
+  private void executePersist(final TriggerKey triggerKey, final Trigger.TriggerState newTriggerState) {
     try {
       factory.createExecutor(ImmutableList.of((TransactionAware) table))
                             .execute(new TransactionExecutor.Subroutine() {
                               @Override
                               public void apply() throws Exception {
                                 if (triggerKey != null) {
-                                  TriggerStatus trigger = readTrigger(triggerKey);
-                                  if (trigger != null) {
-                                    persistTrigger(table, trigger.trigger, state);
+                                  TriggerStatus storedTriggerStatus = readTrigger(triggerKey);
+                                  if (storedTriggerStatus != null) {
+                                    // its okay to persist the same trigger back again since during pause/resume
```

Comment:
this comment is confusing. You are not storing the same trigger again, You are storing with a new state.